packages/ui/src/components/Stepper/index.tsx

```diff
@@ -94,102 +231,139 @@ const StyledContainer = styled.div<{
   }
 `
 
-type StepperNumbersProps = {
-  temporal: Temporal
-  children: ReactNode
-  CurrentStep: number
-  size?: 'small' | 'medium'
-}
-
-const StepperNumbers = ({
-  temporal,
+const Step = ({
+  index = 0,
+  onClick,
+  disabled = false,
+  title,
   children,
-  CurrentStep,
-  size = 'medium',
-}: StepperNumbersProps) => (
-  <StyledStepContainer>
-    <StyledBullet
-      sentiment={temporal === 'next' ? 'neutral' : 'primary'}
-      {...(temporal === 'previous'
-        ? {
-            icon: 'check',
+}: StepProps) => {
+  const currentState = useContext(State)
+  const isActive = index === currentState.step
+  const isDone = index < currentState.step
+
+  const textVariant = useMemo(() => {
+    if (currentState.size === 'medium') {
+      return isActive ? 'bodyStrong' : 'body'
+    }
+
+    return isActive ? 'bodySmallStrong' : 'bodySmall'
+  }, [currentState.size, isActive])
+
+  return (
+    <StyledStepContainer
+      className="step"
+      data-interactive={currentState.interactive && isDone}
+      isActive={isActive}
+      isDone={isDone}
+      onClick={() => {
+        if (currentState.interactive && !disabled) {
+          if (index < currentState.step) {
+            currentState.setStep(index)
           }
-        : {
-            text: CurrentStep.toString(),
-          })}
-      prominence={temporal !== 'current' ? 'default' : 'strong'}
-      size={size}
-    />
-
-    <StyledText
-      as="span"
-      variant={size === 'medium' ? 'body' : 'bodySmall'}
-      prominence={temporal === 'next' ? 'weak' : 'default'}
+          onClick?.(index)
+        }
+      }}
+      data-disabled={disabled}
+      data-testid={`stepper-step-${index}`}
```

Comment:
Here you could say something like `data-testid={dataTestId ?? `stepper-step-${index}`}` after adding data test id